### PR TITLE
[Improve][Connector-V2][TableStore] Unified excetion for TableStore sink connector

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request
https://github.com/apache/incubator-seatunnel/issues/3547
sub-task of https://github.com/apache/incubator-seatunnel/issues/2930
Unified excetion for TableStore sink connector

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by TaoZex** [comment]
@TyrantLucifer @wuchunfu PTAL

**Review by EricJoy2048 — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/serialize/DefaultSeaTunnelRowSerializer.java [hunk 1]
```diff
@@ -92,7 +94,8 @@ private ColumnType convertColumnType(SeaTunnelDataType<?> seaTunnelDataType) {
             case BYTES:
                 return ColumnType.BINARY;
             default:
-                throw new UnsupportedOperationException("Unsupported columnType: " + seaTunnelDataType);
+                throw new TablestoreConnectorException(TablestoreConnectorErrorCode.UNSUPPORTED_COLUMN_TYPE,
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/serialize/DefaultSeaTunnelRowSerializer.java]: CommonErrorCode.UNSUPPORTED_DATA_TYPE is better, please remove this redundant error code.

File: seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/serialize/DefaultSeaTunnelRowSerializer.java [hunk 2]
```diff
@@ -115,7 +118,8 @@ private PrimaryKeyType convertPrimaryKeyType(SeaTunnelDataType<?> seaTunnelDataT
             case BYTES:
                 return PrimaryKeyType.BINARY;
             default:
-                throw new UnsupportedOperationException("Unsupported primaryKeyType: " + seaTunnelDataType);
+                throw new TablestoreConnectorException(TablestoreConnectorErrorCode.UNSUPPORTED_PRIMARY_KEY_TYPE,
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/serialize/DefaultSeaTunnelRowSerializer.java]: CommonErrorCode.UNSUPPORTED_DATA_TYPE is better, please remove this redundant error code.

File: seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/serialize/DefaultSeaTunnelRowSerializer.java [hunk 3]
```diff
@@ -135,7 +139,8 @@ private Column convertColumn(String columnName, Object value, ColumnType columnT
             case BINARY:
                 return new Column(columnName, ColumnValue.fromBinary((byte[]) value));
             default:
-                throw new UnsupportedOperationException("Unsupported columnType: " + columnType);
+                throw new TablestoreConnectorException(TablestoreConnectorErrorCode.UNSUPPORTED_COLUMN_TYPE,
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/serialize/DefaultSeaTunnelRowSerializer.java]: CommonErrorCode.UNSUPPORTED_DATA_TYPE is better, please remove this redundant error code.

File: seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/serialize/DefaultSeaTunnelRowSerializer.java [hunk 4]
```diff
@@ -151,7 +156,8 @@ private PrimaryKeyColumn convertPrimaryKeyColumn(String columnName, Object value
             case BINARY:
                 return new PrimaryKeyColumn(columnName, PrimaryKeyValue.fromBinary((byte[]) value));
             default:
-                throw new UnsupportedOperationException("Unsupported primaryKeyType: " + primaryKeyType);
+                throw new TablestoreConnectorException(TablestoreConnectorErrorCode.UNSUPPORTED_PRIMARY_KEY_TYPE,
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/serialize/DefaultSeaTunnelRowSerializer.java]: CommonErrorCode.UNSUPPORTED_DATA_TYPE is better, please remove this redundant error code.

File: seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/sink/TablestoreSinkClient.java [hunk 5]
```diff
@@ -120,7 +122,8 @@ synchronized void flush() throws IOException {
 
     private void checkFlushException() {
         if (flushException != null) {
-            throw new RuntimeException("Writing items to Tablestore failed.", flushException);
+            throw new TablestoreConnectorException(TablestoreConnectorErrorCode.FLUSH_DATA_FAILED,
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/sink/TablestoreSinkClient.java]: CommonErrorCode.FLUSH_DATA_FAILED is better, please remove this redundant error code.

File: seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/sink/TablestoreSinkClient.java [hunk 6]
```diff
@@ -110,8 +111,9 @@ synchronized void flush() throws IOException {
 
         if (!response.isAllSucceed()) {
             for (BatchWriteRowResponse.RowResult rowResult : response.getFailedRows()) {
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/sink/TablestoreSinkClient.java]: Why using for here?

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

File: seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/serialize/DefaultSeaTunnelRowSerializer.java (hunk 1 above)
Comment (TaoZex) [reply seatunnel-connectors-v2/connector-tablestore/src/main/java/org/apache/seatunnel/connectors/seatunnel/tablestore/serialize/DefaultSeaTunnelRowSerializer.java]: Thank you for your advice. It helped me a lot. I have fixed it.

**Review by TyrantLucifer — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**